apps/web/src/services/dashboards/widgets/_hooks/use-widget-lifecycle-deprecated.ts

```diff
@@ -0,0 +1,232 @@
+        if (disposeWidget) disposeWidget();
+    });
+    watch(() => dashboardDetailState.variables, (after, before) => {
+        if (!props.initiated || props.errorMode || !props.inheritOptions) return;
+        if (dashboardDetailState.widgetViewModeModalVisible && (props.widgetKey !== widgetFormState.widgetKey)) return; // disable when it's view mode
+        const _isRefreshable = checkRefreshableByDashboardVariables(props.inheritOptions, after, before);
+        if (_isRefreshable) refreshWidgetAndUpdateDataMap(refreshWidget, props.widgetKey);
+    }, { deep: true });
+    watch(() => dashboardDetailState.variablesSchema, (after, before) => {
+        if (!props.editMode || !props.inheritOptions || isEqual(after, before)) return;
+        if (dashboardDetailState.widgetViewModeModalVisible && (props.widgetKey !== widgetFormState.widgetKey)) return;
+        if (!state.widgetInfo) return;
+
+        const [
+            addedVariableSchemaProperties,
+            deletedVariableSchemaProperties,
+            changedVariableSchemaProperties,
+        ] = getUpdatedDashboardVariableSchemaProperties(after, before);
+        let isWidgetOptionChanged; let isWidgetOptionAdded; let isWidgetOptionDeleted;
+        if (addedVariableSchemaProperties.length) {
+            isWidgetOptionAdded = updateWidgetByAddedVariableSchemaProperties(addedVariableSchemaProperties, state.widgetInfo, props.widgetKey, state.widgetConfig);
+        }
+        if (deletedVariableSchemaProperties.length) {
+            isWidgetOptionDeleted = updateWidgetByDeletedVariableSchemaProperties(deletedVariableSchemaProperties, state.widgetInfo, props.widgetKey, state.widgetConfig, props.inheritOptions, after);
+        }
+        if (changedVariableSchemaProperties.length) {
+            isWidgetOptionChanged = isAffectedByChangedVariableSchemaProperties(changedVariableSchemaProperties, props.inheritOptions);
+        }
+
+        if (!props.initiated) return;
+        if (isWidgetOptionAdded || isWidgetOptionDeleted || isWidgetOptionChanged) {
+            refreshWidgetAndUpdateDataMap(refreshWidget, props.widgetKey);
+        }
+    }, { deep: true });
+    if (state.settings) {
+        watch(() => state.settings, (current, previous) => {
+            if (!current || !previous) return;
+            if (dashboardDetailState.widgetViewModeModalVisible && (props.widgetKey !== widgetFormState.widgetKey)) return;
+            if (current.date_range.start !== previous.date_range.start || current.date_range.end !== previous.date_range.end) {
+                refreshWidgetAndUpdateDataMap(refreshWidget, props.widgetKey);
+            } else if (onCurrencyUpdate && current?.currency?.value !== previous?.currency?.value) {
+                onCurrencyUpdate();
+            }
+        });
+    }
+
+    if (redrawOnLanguageChange) {
+        try {
+            if (!redrawChart) throw Error('redrawChart is required');
+            watch(() => i18n.locale, async () => {
```

Comment:
nitpick: In out of `vue` file, `i18n.locale` and `i18n.t` is changed to `i18n.global.locale` and `i18n.global.t`.